lib/galaxy/datatypes/binary.py

```diff
@@ -462,6 +467,55 @@ def to_archive(self, trans, dataset, name=""):
         file_paths.append(dataset.metadata.bam_index.file_name)
         return zip(file_paths, rel_paths)
 
+    def get_chunk( self, trans, dataset, offset=0, ck_size=None ):
+        index_file = dataset.metadata.bam_index
```

Comment:
@mvdbeek Do you happen to know if this function has the potential to consume unbounded memory if the bam file is problematic or is it going to always be pretty managable despite the bam file?

Reply:
Yes, bam lines (reads) can become huge. If someone did a genome-against-genome alignment I think this could be problematic.

Reply:
We verified the problem on main wasn't data loaded into peak attributes - sorry for the false alarm. Maybe we should create an issue for known points in Galaxy where the amount of data loaded is unbounded, I don't know.